Add missing test launchers for RpcJitTest and DistAutogradJitTest

xush6528 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#32891 Add missing test launchers for RpcJitTest and DistAutogradJitTest**

- Add DistAutoGradJitTest into fork test launcher
- Add RpcJitTest into fork/spawn test launcher

Differential Revision: [D5785394](https://our.internmc.facebook.com/intern/diff/D5785394/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D5785394/)!

xush6528 committed:
Add missing test launchers for RpcJitTest and DistAutogradJitTest

- Add DistAutoGradJitTest into fork test launcher
- Add RpcJitTest into fork/spawn test launcher

Differential Revision: [D5785394](https://our.internmc.facebook.com/intern/diff/D5785394/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D5785394/)!

[ghstack-poisoned]

kostmo commented:
## :pill: CircleCI build failures summary and remediations
As of commit e34f4c72:


**None of the build failures appear to be your fault**.

* **1/1** recognized as flaky :snowflake:
    * Re-run these jobs?


## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=e34f4c72beef2f67fda6a00430949f794d554e48).

### :snowflake: 1 failure recognized as flaky
The following build failures have been detected as flaky and **may not be your fault**:
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/4394767) caffe2_onnx_py3_6_clang7_ubuntu16_04_build (1/1)
**Step:** "Build" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=93951839) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=93951839)) :snowflake:
<details>
<summary>
<code>$<3>$<3> 9045d41e4a2f: Pushing  2.603GB/2.603GB $<3>$<3> 9045d41e4a2f: Pushing  2.603GB/2.603GB toomanyrequests: Rate exceeded</code>
</summary>

```
$<3>$<3> 54b25e145f42: Preparing   
$<3>$<3> f789043da22d: Preparing   
$<3>$<3> 0375cdd5a603: Preparing   
$<3>$<3> 66b793df475b: Preparing   
$<3>$<3> d800130e2cf5: Preparing   
$<3>$<3> eb5c36289557: Preparing   
$<3>$<3> fb61a0ac20b8: Preparing   
$<3>$<3> bc72fb2e7b74: Preparing   
$<3>$<3> 903669ee7207: Preparing   
$<3>$<3> a5a5f8c62487: Preparing   
9045d41e4a2f: Pushing  2.603GB/2.603GB $<3>$<3> 9045d41e4a2f: Pushing  2.603GB/2.603GB toomanyrequests: Rate exceeded 
```

</details>



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

xush6528 committed:
Update on "Add missing test launchers for RpcJitTest and DistAutogradJitTest"


- Add DistAutoGradJitTest into fork test launcher
- Add RpcJitTest into fork/spawn test launcher

Differential Revision: [D5785394](https://our.internmc.facebook.com/intern/diff/D5785394/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D5785394/)!

[ghstack-poisoned]

xush6528 committed:
Update on "Add missing test launchers for RpcJitTest and DistAutogradJitTest"


- Add DistAutoGradJitTest into fork test launcher
- Add RpcJitTest into fork/spawn test launcher

Differential Revision: [D5785394](https://our.internmc.facebook.com/intern/diff/D5785394/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D5785394/)!

[ghstack-poisoned]

xush6528 committed:
Update on "Add missing test launchers for RpcJitTest and DistAutogradJitTest"


- Add DistAutoGradJitTest into fork test launcher
- Add RpcJitTest into fork/spawn test launcher

Differential Revision: [D5785394](https://our.internmc.facebook.com/intern/diff/D5785394/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D5785394/)!

[ghstack-poisoned]

pritamdamania87 commented:
Code changes look good, although looks like there are a lot of CI failures. Can you fix those?

pritamdamania87 reviewed (changes requested):
Need to fix CI

xush6528 committed:
Update on "Add missing test launchers for RpcJitTest and DistAutogradJitTest"


- Add DistAutoGradJitTest into fork test launcher
- Add RpcJitTest into fork/spawn test launcher

Differential Revision: [D5785394](https://our.internmc.facebook.com/intern/diff/D5785394/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D5785394/)!

[ghstack-poisoned]

xush6528 commented:
The error was 
```py
ModuleNotFoundError: No module named 'torch.testing._internal.distributed.rpc.jit
```

I added a folder, "torch/testing/_internal/distributed/rpc/jit/". 
But didn't add `__init__.py` file to make this folder a Python module. This will fail Python imports.

xush6528 committed:
Update on "Add missing test launchers for RpcJitTest and DistAutogradJitTest"


- Add DistAutoGradJitTest into fork test launcher
- Add RpcJitTest into fork/spawn test launcher

Differential Revision: [D5785394](https://our.internmc.facebook.com/intern/diff/D5785394/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D5785394/)!

[ghstack-poisoned]

xush6528 committed:
Update on "Add missing test launchers for RpcJitTest and DistAutogradJitTest"


- Add DistAutoGradJitTest into fork test launcher
- Add RpcJitTest into fork/spawn test launcher

Differential Revision: [D5785394](https://our.internmc.facebook.com/intern/diff/D5785394/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D5785394/)!

[ghstack-poisoned]

xush6528 committed:
Update on "Add missing test launchers for RpcJitTest and DistAutogradJitTest"


- Add DistAutoGradJitTest into fork test launcher
- Add RpcJitTest into fork/spawn test launcher

Differential Revision: [D5785394](https://our.internmc.facebook.com/intern/diff/D5785394/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D5785394/)!

[ghstack-poisoned]

xush6528 commented:
The error happens on windows build. RPC does not support running on windows. So I should add these added tests to windows test blacklist as well.

xush6528 committed:
Update on "Add missing test launchers for RpcJitTest and DistAutogradJitTest"


- Add DistAutoGradJitTest into fork test launcher
- Add RpcJitTest into fork/spawn test launcher

Differential Revision: [D5785394](https://our.internmc.facebook.com/intern/diff/D5785394/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D5785394/)!

[ghstack-poisoned]

xush6528 committed:
Update on "Add missing test launchers for RpcJitTest and DistAutogradJitTest"


- Add DistAutoGradJitTest into fork test launcher
- Add RpcJitTest into fork/spawn test launcher

Differential Revision: [D5785394](https://our.internmc.facebook.com/intern/diff/D5785394/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D5785394/)!

[ghstack-poisoned]

xush6528 committed:
Update on "Add missing test launchers for RpcJitTest and DistAutogradJitTest"


- Add DistAutoGradJitTest into fork test launcher
- Add RpcJitTest into fork/spawn test launcher

Differential Revision: [D5785394](https://our.internmc.facebook.com/intern/diff/D5785394/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D5785394/)!

[ghstack-poisoned]

zhaojuanmao commented on test/distributed/rpc/jit/test_dist_autograd_spawn.py:
```diff
@@ -0,0 +1,21 @@
+@unittest.skipIf(
+    TEST_WITH_ASAN, "Skip ASAN as torch + multiprocessing spawn have known issues"
+)
+class JitDistAutogradTestWithSpawn(MultiProcessTestCase, JitDistAutogradTest):
```
nit: could it be named as "DistAutogradJitTestWithSpawn"?

xush6528 replied:
That sounds more semantic correct. 
But I am thinking about convenience for searching. If we type "DistAutograd", every time we have to choose between autocomplete options, "JitTest" and "Test".

zhaojuanmao commented on torch/testing/_internal/distributed/rpc/jit/dist_autograd_test.py:
```diff
@@ -0,0 +1,38 @@
+@unittest.skipIf(
+    not torch._six.PY3, "Pytorch distributed autograd package does not support python2"
+)
+class JitDistAutogradTest(RpcAgentTestFixture):
```
there are test_trainer_ps_torchscript_functions() and test_backward_simple_script_call() as well , those are hard to move though

xush6528 replied:
I will leave them there for now. This PR is asked from https://github.com/pytorch/pytorch/pull/33329#discussion_r380962331.

zhaojuanmao reviewed (commented):
overall looks good to me, some nits

zhaojuanmao commented on test/distributed/rpc/jit/test_rpc_spawn.py:
```diff
@@ -0,0 +1,19 @@
+@unittest.skipIf(
+    TEST_WITH_ASAN, "Skip ASAN as torch + multiprocessing spawn have known issues"
+)
+class RpcJitTestWithSpawn(MultiProcessTestCase, JitRpcTest):
```
nit: JitRpcTestWithSpawn

xush6528 replied:
Fixed it.

xush6528 committed:
Update on "Add missing test launchers for RpcJitTest and DistAutogradJitTest"


- Add DistAutoGradJitTest into fork test launcher
- Add RpcJitTest into fork/spawn test launcher

Differential Revision: [D5785394](https://our.internmc.facebook.com/intern/diff/D5785394/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D5785394/)!

[ghstack-poisoned]

xush6528 committed:
Update on "Add missing test launchers for RpcJitTest and DistAutogradJitTest"


- Add DistAutoGradJitTest into fork test launcher
- Add RpcJitTest into fork/spawn test launcher

Differential Revision: [D5785394](https://our.internmc.facebook.com/intern/diff/D5785394/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D5785394/)!

[ghstack-poisoned]

zhaojuanmao reviewed (approved):
please make sure tests passed before landing

xush6528 committed:
Update on "Add missing test launchers for RpcJitTest and DistAutogradJitTest"


- Add DistAutoGradJitTest into fork test launcher
- Add RpcJitTest into fork/spawn test launcher

Differential Revision: [D5785394](https://our.internmc.facebook.com/intern/diff/D5785394/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D5785394/)!

[ghstack-poisoned]
